### DISCUSSION TITLE
liburing items to be done for 2.3 release

### DISCUSSION THREAD

**Discussion by axboe** [axboe/liburing#696] [General]
1. ~~zc test fix (@isilence )~~
2. ~~buf ring man page fix (@DylanZA )~~
3. ~~https://github.com/axboe/liburing/issues/659 (man page, @axboe)~~
4. ~~@metze-samba's liburing patch (@axboe, need to get to that)~~
5. ~~io_uring_register_file_alloc_range.3 man page (@isilence)~~

Feel free to update - let's target a release mid next week, giving us all a bit of time to fix items on the list and add anything else that may be needed.

**Comment by ammarfaizi2** [comment]
5. ~~Clean up `-Wshorten-64-to-32` warnings in the main library (@ammarfaizi2).~~

**Comment by axboe** [comment]
6. https://github.com/axboe/liburing/issues/701 (@isilence)

Not sure if we strictly need it for 2.3, but it is an oversight that these aren't documented at all. Particularly as they aren't necessarily intuitive ("why would I need a tag for registered items?").

**Reply by ammarfaizi2** [reply]
> 6\. [io_uring register tags (files and buffers) man pages missing #701](https://github.com/axboe/liburing/issues/701) (@isilence)

Are you still waiting for this?

**Reply by axboe** [reply]
Ideally yes - it's functionality added after 2.2, but there are no man pages for it...

**Comment by ammarfaizi2** [comment]
~~7. Update CHANGELOG file (@axboe).~~

I think this file probably should be updated right when new a feature
is added. No need to wait for a new release. Otherwise, we may forget
it, or mess up the chronological order.

What do you think?

**Reply by axboe** [reply]
Yeah agree, it should be updated as things land. But obviously it did not. Outside of just an itemized changelog, would also be nice to have more of a "This is what's new in liburing 2.x" with a release, highlighting things that may need a bit more of an explanation. Something that aims to be useful for developers of existing and new applications.

**Reply by axboe** [reply]
I filled in the basics of the CHANGELOG

**Comment by axboe** [comment]
2.3 has been released, will open a new one for 2.4 once we get closer to release of that.
